sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/TriggerBinding/EventGridListener.cs

```diff
@@ -11,16 +11,20 @@ internal class EventGridListener : Host.Listeners.IListener
     {
         public ITriggeredFunctionExecutor Executor { get; private set; }
         public readonly bool SingleDispatch;
+        public bool HasEventGridEvent { get; }
```

Comment:
These seem like they may be mutually exclusive, in practice. If I'm not mistaken on that, should we consider a single enum that represents the potential event type bindings instead of individual flags?

Reply:
They are mutually exclusive, but they are not exhaustive, i.e. they can both be false.

Reply:
But your point still stands.